nagios_plugins/check_proc_cpu/check_proc_cpu.sh

```diff
@@ -61,6 +28,63 @@ else
     echo "check_proc_cpu.sh -w 200 -c 400 --cmdpattern \"tomcat7.*java.*Dcom\""
     echo ""
     echo "Copyright (C) 2015 DennyZhang (contact@dennyzhang.com)"
-    exit 3
+}
+
+while [ "$#" -gt 0 ]
+do
+    opt=$1
+    case $opt in
+        -w)
+            warn_cpu=$2
+            shift 2 # Past argument and value
+        ;;
+        -c)
+            criti_cpu=$2
+            shift 2 # Past argument and value
+        ;;
+        --pidfile)
+            pidfile=$2
+            pid=$(cat "$pidfile")
+            shift 2 # Past argument and value
+        ;;
+        --cmdpattern)
+            cmdpattern=$2
+            pid=$(pgrep -a -f "$cmdpattern" | grep -v `basename $0` | head -n 1 | awk -F' ' '{print $1}')
+            shift 2 # Past argument and value
+        ;;
+        --pid)
+            pid=$2
+            shift 2 # Past argument and value
+        ;;
+        *)
+            print_help
+            exit 3
+        ;;
+    esac
+done
+
+num_re='^[0-9]+$'
+if ! [[ "$warn_cpu" =~ $num_re ]] || ! [[ "$criti_cpu" =~ $num_re ]]
+then
+    echo "ERROR: Warning or Critical level is not a number"
+	exit 3
```

Comment:
Would you like to fix the indentation?

Reply:
Oh, I thought I already fixed it. I will.

Reply:
I fixt the rest of them, too.